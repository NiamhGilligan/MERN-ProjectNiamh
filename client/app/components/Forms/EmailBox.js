

import React from 'react';
import EmailField from './EmailField.js';
import NameField from './NameField.js';
import request from 'superagent';
import callback from 'superagent';
import createReactClass from 'create-react-class';


var EmailBox = createReactClass({

  getInitialState: function(){
    return {submitted: false};
  },

  onSubmit: function(e) {
    if (!this.refs.fieldEmail.state.valid) {
      alert("Email is required to sign up!");
    } else {
      // super agent post request where request is sent to server
      request.post('/api/account/emailsignup')
      //url for post request matching server post request url
      //.send sending the users input from email and firstname
        .send({ email: this.refs.fieldEmail.state.value })
        .send({ firstName: this.refs.fieldName.state.value })
        .then(json => {
          if (!json.success) {
            alert('Congratulations! You are registered.');
            window.location.reload()
          }

          if(json.success) {
            alert("Congratulations! You are registered.");
            console.log(subscriber['firstName'] + " ; " + subscriber['email']);
          }
        })
        .then(callback);

      var subscriber = {
        email: this.refs.fieldEmail.state.value,
        firstName: this.refs.fieldName.state.value

      };

      this.refs.fieldEmail.clear();
      this.refs.fieldName.clear();

    }
  },
  onChange: function(msg){
    console.log(msg);
    this.setState({submitted: true});
  },

  render: function() {
    var successStyle = {
      color:"green",
      visibility: this.state.submitted ? "visible" : "hidden"
    };
    var boxStyle ={
        opacity: '0.9',
      float:'right',


    };

    return (
      <div className="col-sm-3" style ={boxStyle}>
        <div className="panel panel-default">
          <div className="panel-heading">
            <h4>Get the latest Lessons sent to you</h4>
          </div>
          <div className="panel-body">
          </div>
          <NameField type="First" ref="fieldName" />
          <br/>
          <EmailField ref="fieldEmail" />
          <div className="row">
            <div className="col-sm-6">
              <button className="btn btn-primary" onClick={this.onSubmit}>
                Submit
              </button>
            </div>
            <div className="col-sm-2">
              <h5 style = {successStyle}>Success</h5>
            </div>
          </div>

        </div>
      </div>
    );
  }

});

export default EmailBox;
