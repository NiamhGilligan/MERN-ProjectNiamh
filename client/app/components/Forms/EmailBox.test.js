import React from 'react';
import {shallow} from 'enzyme';
import EmailBox from './EmailBox';

describe('Email Box Test', ()=> {

  test('Rendering EmailBox page',() => {
    const wrapper = shallow(
      <EmailBox/>
    );

    expect(wrapper).toMatchSnapshot();
  });

});
