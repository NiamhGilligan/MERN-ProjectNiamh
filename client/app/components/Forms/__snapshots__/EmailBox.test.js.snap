// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Box Test Rendering EmailBox page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmailBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="panel panel-default"
      >
        <div
          className="panel-heading"
        >
          <h4>
            Get the latest Lessons sent to you
          </h4>
        </div>
        <div
          className="panel-body"
        />
        <NameField
          type="First"
        />
        <br />
        <EmailField />
        <div
          className="row"
        >
          <div
            className="col-sm-6"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Submit
            </button>
          </div>
          <div
            className="col-sm-2"
          >
            <h5
              style={
                Object {
                  "color": "green",
                  "visibility": "hidden",
                }
              }
            >
              Success
            </h5>
          </div>
        </div>
      </div>,
      "className": "col-sm-3",
      "style": Object {
        "float": "right",
        "opacity": "0.9",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="panel-heading"
          >
            <h4>
              Get the latest Lessons sent to you
            </h4>
          </div>,
          <div
            className="panel-body"
          />,
          <NameField
            type="First"
          />,
          <br />,
          <EmailField />,
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
              >
                Submit
              </button>
            </div>
            <div
              className="col-sm-2"
            >
              <h5
                style={
                  Object {
                    "color": "green",
                    "visibility": "hidden",
                  }
                }
              >
                Success
              </h5>
            </div>
          </div>,
        ],
        "className": "panel panel-default",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
              Get the latest Lessons sent to you
            </h4>,
            "className": "panel-heading",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Get the latest Lessons sent to you",
            },
            "ref": null,
            "rendered": "Get the latest Lessons sent to you",
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "panel-body",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "type": "First",
          },
          "ref": "fieldName",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": "fieldEmail",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-sm-6"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  Submit
                </button>
              </div>,
              <div
                className="col-sm-2"
              >
                <h5
                  style={
                    Object {
                      "color": "green",
                      "visibility": "hidden",
                    }
                  }
                >
                  Success
                </h5>
              </div>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  Submit
                </button>,
                "className": "col-sm-6",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "btn btn-primary",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h5
                  style={
                    Object {
                      "color": "green",
                      "visibility": "hidden",
                    }
                  }
                >
                  Success
                </h5>,
                "className": "col-sm-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Success",
                  "style": Object {
                    "color": "green",
                    "visibility": "hidden",
                  },
                },
                "ref": null,
                "rendered": "Success",
                "type": "h5",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel panel-default"
        >
          <div
            className="panel-heading"
          >
            <h4>
              Get the latest Lessons sent to you
            </h4>
          </div>
          <div
            className="panel-body"
          />
          <NameField
            type="First"
          />
          <br />
          <EmailField />
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
              >
                Submit
              </button>
            </div>
            <div
              className="col-sm-2"
            >
              <h5
                style={
                  Object {
                    "color": "green",
                    "visibility": "hidden",
                  }
                }
              >
                Success
              </h5>
            </div>
          </div>
        </div>,
        "className": "col-sm-3",
        "style": Object {
          "float": "right",
          "opacity": "0.9",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="panel-heading"
            >
              <h4>
                Get the latest Lessons sent to you
              </h4>
            </div>,
            <div
              className="panel-body"
            />,
            <NameField
              type="First"
            />,
            <br />,
            <EmailField />,
            <div
              className="row"
            >
              <div
                className="col-sm-6"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                >
                  Submit
                </button>
              </div>
              <div
                className="col-sm-2"
              >
                <h5
                  style={
                    Object {
                      "color": "green",
                      "visibility": "hidden",
                    }
                  }
                >
                  Success
                </h5>
              </div>
            </div>,
          ],
          "className": "panel panel-default",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4>
                Get the latest Lessons sent to you
              </h4>,
              "className": "panel-heading",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Get the latest Lessons sent to you",
              },
              "ref": null,
              "rendered": "Get the latest Lessons sent to you",
              "type": "h4",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "panel-body",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "type": "First",
            },
            "ref": "fieldName",
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": "fieldEmail",
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-sm-6"
                >
                  <button
                    className="btn btn-primary"
                    onClick={[Function]}
                  >
                    Submit
                  </button>
                </div>,
                <div
                  className="col-sm-2"
                >
                  <h5
                    style={
                      Object {
                        "color": "green",
                        "visibility": "hidden",
                      }
                    }
                  >
                    Success
                  </h5>
                </div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn btn-primary"
                    onClick={[Function]}
                  >
                    Submit
                  </button>,
                  "className": "col-sm-6",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit",
                    "className": "btn btn-primary",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h5
                    style={
                      Object {
                        "color": "green",
                        "visibility": "hidden",
                      }
                    }
                  >
                    Success
                  </h5>,
                  "className": "col-sm-2",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Success",
                    "style": Object {
                      "color": "green",
                      "visibility": "hidden",
                    },
                  },
                  "ref": null,
                  "rendered": "Success",
                  "type": "h5",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
